src/e4stypes/book_item.py

```diff
@@ -22,3 +31,26 @@ def get_edition(self) -> int:
 
     def get_course_number(self) -> int:
         return self._course_number
+
+    @classmethod
+    def add_book(cls, title: str, desc: str, price: Decimal, weight: float,
+                 seller: str, book_title: str, edition: int,
+                 course_number: int) -> Book:
```

Comment:
all of these parameters are already handled when creating a `BookItem`. Instead of passing each field of `BookItem` as a parameter here, we should just pass a `BookItem` and access the fields while creating the dictionary. Same will be true for each other variation of `add_item`.

More of a nitpick, but also if we call these all `add_item()` we may be able to utilize polymorphism.

```suggestion
    def add_item(cls, title: str, desc: str, price: Decimal, weight: float,
                 seller: str, book_title: str, edition: int,
                 course_number: int) -> BookItem:
```

Reply:
Makes sense, would the only thing I need to change in these functions are the changes you've annotated? Or is there something more to this that I just didn't understand? @carpenterd777

Reply:
you'd have to change the design of the method a bit. instead of being passed each of these parameters, the method would receive a `BookItem` that already exists. e.g.:
```
def add_item(cls, book : BookItem) -> BookItem:
```

then when creating the dictionary to send to mongo you'd access the fields of `book` either directly or using their getter methods.